### PR TITLE
Fix errors made in merged PR #271 (commit b568955)

### PR DESCRIPTION
**Pull request by dgsga**
This PR fixes errors made when creating PR #271. Please let me know if you would like to revert commit b568955 and have me prepare a fresh full PR or are happy just to apply this PR. I have compiled this fix in an Archlinux VM and all is well. Compiles fine in my fork, afp.conf is read correctly and shared volumes mount successfully.

### CODE REVIEW

File: libatalk/iniparser/dictionary.c [hunk 1]
```diff
@@ -34,31 +34,27 @@ DEALINGS IN THE SOFTWARE.
 /*--------------------------------------------------------------------------*/
 
 /*---------------------------------------------------------------------------
-   								Includes
+                                                                Includes
  ---------------------------------------------------------------------------*/
-#ifdef HAVE_CONFIG_H
 #include "config.h"
-#endif /* HAVE_CONFIG_H */
```
Comment (rdmark) [inline libatalk/iniparser/dictionary.c:41]: Any particular reason for removing this ifdef here? I think HAVE_CONFIG_H is still used in most other places in the codebase.

File: libatalk/iniparser/iniparser.c [hunk 2]
```diff
@@ -29,18 +29,16 @@ DEALINGS IN THE SOFTWARE.
 */
 
 /*---------------------------- Includes ------------------------------------*/
-#ifdef HAVE_CONFIG_H
 #include "config.h"
-#endif /* HAVE_CONFIG_H */
```
Comment (rdmark) [inline libatalk/iniparser/iniparser.c:34]: Same as above. I do think it's safe to remove this, but it should be done so consistently across the codebase in one go, IMHO.

**Comment by dgsga** [comment]
Okeydoke. Will fix the two ifdefs and force push the changes to the iniparser branch tomorrow AM

**Comment by rdmark** [comment]
Sounds good. Also, can you please make the commit message a bit more descriptive? See also: commit message guidelines on the wiki: https://github.com/Netatalk/netatalk/wiki/Developer-Notes#user-content-Commit_messages

**Comment by rdmark** [comment]
This code produces the following warnings in my Debian Bullseye / arm / gcc10 environment.
```
make[3]: Entering directory '/home/dmark/dev/netatalk/libatalk/iniparser'
  CC       dictionary.lo
  CC       iniparser.lo
dictionary.c: In function ‘makekey’:
dictionary.c:65:5: warning: implicit declaration of function ‘strlcpy’; did you mean ‘strncpy’? [-Wimplicit-function-declaration]
   65 |     strlcpy(buf, section, MAXKEYSIZE);
      |     ^~~~~~~
      |     strncpy
dictionary.c:67:9: warning: implicit declaration of function ‘strlcat’; did you mean ‘strncat’? [-Wimplicit-function-declaration]
   67 |         strlcat(buf, ":", MAXKEYSIZE);
      |         ^~~~~~~
      |         strncat
  CCLD     libiniparser.la
make[3]: Leaving directory '/home/dmark/dev/netatalk/libatalk/iniparser'
```

A quick googling suggests that `strlcpy` and `strlcat` aren't part of Linux libc. [This one workaround](https://unix.stackexchange.com/questions/230948/implicit-declaration-of-strlcpy-and-strlcat-even-with-string-h-included) worked for me but I wonder if there's a way to not introduce a dependency on BSD libc headers?

**Comment by dgsga** [comment]
Ifdefs have been added back. Many of the string functions seem to have their cons as well as pro's, they are deprecated in some operating systems but not in others which leads to an inconsistent picture. Not sure where to go with this one, you could always revert the iniparser commit if you'd rather leave things as they were.

**Comment by rdmark** [comment]
Never mind. The calls to strlcpy and strlcat were there even before your changesets. We can deal with the portability improvements elsewhere. Or perhaps someone more well versed in libc portability can tell us that it is a non-problem. :)

**Comment by rdmark** [comment]
BTW, with "more descriptive" commit message, I meant something that objectively and independently describes the substance of the changeset. In this case, perhaps something like "add missing logic for iniparser 4.1 upgrade, and code style improvements"

**Pull request merged**